### PR TITLE
[Native] [API] Authz docs consolidation

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*

*Description of changes:*
- consolidate authz into the common file for android/ios
- use fragments for code snippets
- removed references to AWSMobileClient and awsconfiguration.json and sdk links
- TODO android OIDC code snippet

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/graphqlapi/fragments/android/authz/20_oidc.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+Add the following code to your app:
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+// TODO
+public class MyOidcURLRequestInterceptor: URLRequestInterceptor {
```
Comment (lawmicha) [inline docs/lib/graphqlapi/fragments/android/authz/20_oidc.md:8]: looking for some help here to fill in the java/kotlin's interceptor, @jpignata @raphkim ?

File: docs/lib/graphqlapi/fragments/android/authz/10_userpool.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+```java
+Amplify.addPlugin(new AWSCognitoAuthPlugin());
+Amplify.addPlugin(new AWSApiPlugin());
+```
+
+</amplify-block>
+<amplify-block name="Kotlin">
+
+```kotlin
+Amplify.addPlugin(AWSCognitoAuthPlugin());
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/android/authz/10_userpool.md:15]: Remove semi-colon.

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 3]
```diff
@@ -1,9 +1,29 @@
 For client authorization AppSync supports API Keys, Amazon IAM credentials, Amazon Cognito User Pools, and 3rd party OIDC providers. This is inferred from the `amplifyconfiguration.json` file when you call `Amplify.configure()`.
 
-## API Key
+#### API Key
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:3]: API key (no reason to title case this as it isn't a proper noun)

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
 
-API Key is the easiest way to setup and prototype your application with AppSync.
+API Key is the easiest way to setup and prototype your application with AppSync. 
+
+#### Cognito User Pools
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:7]: User Pools and Identity Pools are actually lower cased in their docs. By convention, refer to the service as Amazon Cognito in every mention.

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 5]
```diff
@@ -1,9 +1,29 @@
...
+
+Amazon Cognito User Pools is the most common service to use with AppSync when adding user Sign-Up and Sign-In to your application. If your application needs to interact with other AWS services besides AppSync, such as S3, you will need to use IAM credentials with Cognito Identity Pools. The Amplify CLI can automatically configure this for you when running `amplify add auth` and can also automatically federate User Pools with Identity Pools. This allows you to have both User Pool credentials for AppSync and AWS credentials for S3. You can learn more about Amplify Auth outlined in the [Accessing credentials section](~/lib/auth/access_credentials.md). For manual configuration, add the following snippet to your `amplifyconfiguration.json` file, under the `awsCognitoAuthPlugin`:
```
Comment (jpignata) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:9]: By convention, refer to AWS AppSync by its full name in every mention.

Comment (jpignata) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:9]: > and can also automatically federate User Pools with Identity Pools

This is a tough clause and I'm not sure it explains itself fully. Can we simplify the language and express the concept more directly?

File: docs/lib/graphqlapi/fragments/android/authz/20_oidc.md (hunk 1 above)
Comment (jpignata) [reply docs/lib/graphqlapi/fragments/android/authz/20_oidc.md:8]: @richardmcclellan @raphkim I have no idea haha. Can you weigh in here?

Comment (raphkim) [reply docs/lib/graphqlapi/fragments/android/authz/20_oidc.md:8]: Android doesn't yet allow custom interceptors, but we are looking to add that in soon. 
At the moment, the customer must implement `OidcAuthProvider` interface to return a token, and our interceptor factory will know how to generate an interceptor for that.

Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/authz/20_oidc.md:8]: is that something we want to advise the customer in the meantime or should we leave this section out of Android for now?

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md (hunk 3 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/native_common/authz/common.md:3]:
```suggestion
#### API key
```

File: docs/lib/graphqlapi/fragments/android/authz/10_userpool.md (hunk 2 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/authz/10_userpool.md:15]:
```suggestion
Amplify.addPlugin(AWSCognitoAuthPlugin())
```

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md (hunk 4 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/native_common/authz/common.md]: updated all places to use `user pools` and `identity pools`

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md (hunk 5 above)
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/native_common/authz/common.md]: took a stab at this...
```
Amplify CLI can automatically configure this for you when running `amplify add auth` and will also automatically use the authenticated user from user pools to federate with the identity pools to provide the AWS IAM credentials in the application. [See this for more information about the differences](https://aws.amazon.com/premiumsupport/knowledge-center/cognito-user-pools-identity-pools/)
```

File: docs/lib/graphqlapi/fragments/android/authz/20_oidc.md (hunk 1 above)
Comment (raphkim) [reply docs/lib/graphqlapi/fragments/android/authz/20_oidc.md:8]: i will make a commit to this branch with the instructions for adding OIDC auth provider to the category

File: docs/lib/graphqlapi/fragments/android/authz/30_multi.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+You can make a call to the specific client using the friendly name:
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+Amplify.API.mutate(
+    "friendly_name_API_KEY",
```
[the remark this answers is not in the document]
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/authz/30_multi.md]: will update to all places to square bracket and capitalized casing. like `[OPEN-ID-CONNECT-TOKEN]`

[the remark this answers is not in the document]
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/authz/30_multi.md]: updated for consistency FRIENDLY-NAME-API-WITH-API-KEY

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 7]
```diff
@@ -1,9 +1,29 @@
 For client authorization AppSync supports API Keys, Amazon IAM credentials, Amazon Cognito User Pools, and 3rd party OIDC providers. This is inferred from the `amplifyconfiguration.json` file when you call `Amplify.configure()`.
 
-## API Key
+#### API key
 
-API Key is the easiest way to setup and prototype your application with AppSync.
+API Key is the easiest way to setup and prototype your application with AppSync.
```
[the remark this answers is not in the document]
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/native_common/authz/common.md]: overall improved wording, however left out some details regarding production uses. auth directives allow ways to use apikey in conjunction to the other auth modes so perhaps there's a production use case where we expose APIKey for the public interface. However they can alternatively use IAM Auth with Amplify.Auth to do the sigV4 signing on requests for unauthenticated access. So i'm not completely sure the official stance on it

File: docs/lib/graphqlapi/fragments/android/authz/30_multi.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
+You can make a call to the specific client using the friendly name:
```
[the remark this answers is not in the document]
Comment (lawmicha) [reply docs/lib/graphqlapi/fragments/android/authz/30_multi.md]: updated to the suggested wording

**Pull request merged**